core/src/main/java/org/elasticsearch/common/settings/Settings.java

```diff
@@ -81,7 +81,7 @@ public static boolean getSettingsRequireUnits() {
     private transient ClassLoader classLoader;
 
     Settings(Map<String, String> settings, ClassLoader classLoader) {
-        this.settings = ImmutableMap.copyOf(settings);
+        this.settings = ImmutableMap.copyOf(new TreeMap<>(settings));
```

Comment:
I changed this to ImmutableSortedMap.copyOf, and added a comment that this is needed for consistent serialization. In the future, we can change this to Collections.unmodifiableMap()